src/gui_common/menus/PlanetCustomizerTool.cs

```diff
@@ -0,0 +1,121 @@
+    [Export]
+    private PatchMapDrawer patchMapDrawer = null!;
+
+    [Export]
+    private PatchDetailsPanel patchDetailsPanel = null!;
+
+    [Export]
+    private PanelContainer settingsPanel = null!;
+
+    [Export]
+    private Control patchMapPanel = null!;
+
+    [Export]
+    private Control patchMapButtons = null!;
+
+    [Export]
+    private Button generateButton = null!;
+
+    [Export]
+    private PlanetSettings planetSettings = null!;
+#pragma warning restore CA2213
+
+    private WorldGenerationSettings worldSettings = null!;
+
+    private GameProperties gameProperties = null!;
+
+    public override void _Ready()
+    {
+        base._Ready();
+        InitNewWorld(SimulationParameters.Instance.AutoEvoConfiguration);
+        patchMapDrawer.OnSelectedPatchChanged += UpdatePatchDetailPanel;
+    }
+
+    private void InitNewWorld(IAutoEvoConfiguration configuration)
+    {
+        var planetGenerationSettings = planetSettings.GetPlanetSettings();
+        worldSettings = new WorldGenerationSettings
+        {
+            AutoEvoConfiguration = configuration,
+            WorldSize = planetGenerationSettings.WorldSize,
+            WorldTemperature = planetGenerationSettings.WorldTemperature,
+            WorldSeaLevel = planetGenerationSettings.WorldSeaLevel,
+            GeologicalActivity = planetGenerationSettings.GeologicalActivity,
+            ClimateInstability = planetGenerationSettings.ClimateInstability,
+            Origin = planetGenerationSettings.Origin,
+            DayNightCycleEnabled = planetGenerationSettings.DayNightCycleEnabled,
+            DayLength = planetGenerationSettings.DayLength,
+            LAWK = planetGenerationSettings.LAWK,
+        };
+
+        gameProperties = GameProperties.StartNewMicrobeGame(worldSettings);
+        gameProperties.GameWorld.Map.RevealAllPatches();
+        patchMapDrawer.PlayerPatch = null;
+        patchMapDrawer.Map = gameProperties.GameWorld.Map;
+        patchMapDrawer.SelectedPatch = patchMapDrawer.PlayerPatch;
+        patchDetailsPanel.SelectedPatch = patchMapDrawer.PlayerPatch;
+    }
+
+    private void UpdatePatchDetailPanel(PatchMapDrawer drawer)
+    {
+        var selectedPatch = drawer.SelectedPatch;
+
+        if (selectedPatch == null)
+            return;
+
+        patchDetailsPanel.SelectedPatch = selectedPatch;
+    }
+
+    private void OnBackButtonPressed()
+    {
+        TransitionManager.Instance.AddSequence(ScreenFade.FadeType.FadeOut, 0.1f,
+            SceneManager.Instance.ReturnToMenu, false);
+    }
+
+    private void OnGeneratePressed()
+    {
+        if (settingsPanel.Visible)
+        {
+            InitNewWorld(SimulationParameters.Instance.AutoEvoConfiguration);
+            generateButton.Text = Localization.Translate("BACK_TO_SETTINGS");
+        }
+        else
+        {
+            generateButton.Text = Localization.Translate("GENERATE_BUTTON");
+        }
+
+        settingsPanel.Visible = !settingsPanel.Visible;
```

Comment:
Could you add a comment explaining this line of code? I've read this entire file and I just can't reason why this toggling of visibility is here and specifically after the button text has been updated.

Reply:
Wouldn't it make more sense if the visibility state was changed *first* and then the current data applied to the button text etc? That way this would look like a toggle button callback and then easy enough to understand applying of current state to the GUI.